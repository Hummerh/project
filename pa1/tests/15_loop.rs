fn foo() -> ! {
      let x = 0;
      loop {
            x + 1;
      }
}

fn main() {
    let mut count = 32;

    // Infinite loop
    loop {
        count += 1;

        if count == 3 {
            // Skip the rest of this iteration
            continue;
        }

        if count == 5 {
            // Exit this loop
            break;
        }
    }

    let mut i = 0;
    while i < 10 {
      i += 1;
    }
}
