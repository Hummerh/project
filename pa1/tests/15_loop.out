fn
ID(foo)
(
)
->
!
{
let
ID(x)
=
LITDEC(0)
;
loop
{
ID(x)
+
LITDEC(1)
;
}
}
fn
ID(main)
(
)
{
let
mut
ID(count)
=
LITDEC(32)
;
loop
{
ID(count)
+=
LITDEC(1)
;
if
ID(count)
==
LITDEC(3)
{
continue
;
}
if
ID(count)
==
LITDEC(5)
{
break
;
}
}
let
mut
ID(i)
=
LITDEC(0)
;
while
ID(i)
<
LITDEC(10)
{
ID(i)
+=
LITDEC(1)
;
}
}
